webui/react/src/components/ExperimentActionDropdown.tsx

```diff
@@ -98,6 +104,17 @@ const ExperimentActionDropdown: React.FC<Props> = ({
           openCommand(tensorboard);
           break;
         }
+        case Action.Pin: {
+          const pinList = [ ...settings.pinned ];
+          pinList.push(id);
+          updateSettings({ pinned: [ ...Array.from(new Set(pinList)) ] });
+          break;
+        }
+        case Action.Unpin: {
+          const pinList = settings.pinned.filter((p) => p !== id);
+          updateSettings({ pinned: [ ...Array.from(new Set(pinList)) ] });
```

Comment:
perf: Since you're going to be converting it into a Set anyway to dedupe the Array, it would be faster to do something like this:
```
const pinList = new Set(settings.pinned);
pinList.delete(id);
updateSettings({ pinned: [ ...pinList ] });
```
Benchmarked here: https://jsben.ch/zXESV

Reply:
thats a nice website. 
I agree. I worked on it at night. My brain was dead :( 
Thanks for the pointing out